apps/vanilla/src/pages/exercise1/style/components.css

```
@@ -112,6 +123,25 @@
   border-color: var(--on-surface-color);
 }
 
+/* Chrome, Firefox, Opera, Safari 10.1+ */
+.search-input input::placeholder {
+  color: var(--on-surface-dim-color);
+
+  /* Firefox */
+  opacity: 1;
+}
+
+/* It need for solve problem with placeholder in Microsoft Edge, Internet Explorer 10-11 */
+/* stylelint-disable-next-line selector-pseudo-class-no-unknown */
+.search-input input:input-placeholder {
+  color: var(--on-surface-dim-color);
+}
+
+/* stylelint-disable-next-line selector-pseudo-element-no-unknown */
+.search-input input::input-placeholder { /* Microsoft Edge */
+  color: var(--on-surface-dim-color);
+}
+
```

If they have the same styles, you can reduce your code 
```suggestion
/* It need for solve problem with placeholder in Microsoft Edge, Internet Explorer 10-11 */
/* stylelint-disable-next-line selector-pseudo-class-no-unknown */
.search-input input:input-placeholder,
.search-input input::input-placeholder {
  color: var(--on-surface-dim-color);
}
```